design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -0,0 +1,21 @@
+type ConfigurableProduct {
+    configurable_options_selection_metadata(selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Metadata for the specified configurable options selection")
+}
+
+type ConfigurableOptionsSelectionMetadata @doc(description: "Metadata corresponding to the configurable options selection.")
+{
+    options_available_for_selection: [ConfigurableOptionAvailableForSelection!] @doc(description: "Configurable options available for further selection based on current selection.")
+    media_gallery: [MediaGalleryInterface!] @doc(description: "Product images and videos corresponding to the specified configurable options selection.")
+    variant: SimpleProduct @doc(description: "Variant represented by the specified configurable options selection. It is expected to be null, until selections are made for each configurable option.")
+}
```

Comment:
- ~~Shouldn’t `is_available_for_selection` also be available here? If I select a certain color it can be that resulting sizes aren’t all selectable?~~ That is available bij the `available_value_ids`
- Shouldn't `price_range` be available here as well or do you expect to that recalculation on the client (seems cumbersome)?

I like the possible extension point here to add delivery date information!

Reply:
Let's discuss in the comments above since it is all related: https://github.com/magento/architecture/pull/394#discussion_r456471471